lib/maths/CPeriodicityHypothesisTests.cc

```diff
@@ -1376,167 +1587,224 @@ bool CPeriodicityHypothesisTests::testPeriod(const TTimeTimePr2Vec& windows,
 
     LOG_TRACE(<< "Testing period " << period_);
 
-    if (!this->testStatisticsFor(buckets, stats) || stats.nullHypothesisGoodEnough()) {
+    if (this->testStatisticsFor(buckets, stats) == false ||
+        stats.nullHypothesisGoodEnough()) {
         return false;
     }
     if (stats.s_HasPeriod) {
         stats.s_R0 = stats.s_Rt;
         return true;
     }
 
+    // Compute the effective period in buckets.
     period_ = std::min(period_, length(windows[0]));
     std::size_t period{static_cast<std::size_t>(period_ / m_BucketLength)};
 
     // We need to observe a minimum number of repeated values to test with
     // an acceptable false positive rate.
-    if (!this->seenSufficientPeriodicallyPopulatedBucketsToTest(buckets, period)) {
+    if (this->seenSufficientPeriodicallyPopulatedBucketsToTest(buckets, period) == false) {
         return false;
     }
 
     core_t::TTime windowLength{length(windows)};
     TTimeTimePr2Vec window{{0, windowLength}};
+
+    // Get the values minus the trend for the null hypothesis.
     TFloatMeanAccumulatorVec values(buckets.begin(), buckets.end());
     this->conditionOnHypothesis(stats, values);
 
+    // If necessary project the set of all buckets onto the set which
+    // intersect the windows.
     if (windowLength < length(buckets, m_BucketLength)) {
         LOG_TRACE(<< "Projecting onto " << core::CContainerPrinter::print(windows));
         TFloatMeanAccumulatorVec projection;
         project(values, windows, m_BucketLength, projection);
         values = std::move(projection);
     }
 
-    double B{static_cast<double>(std::count_if(
+    // Compute the number of non-empty buckets. This needs to account for
+    // projecting onto windows.
+    stats.s_B = static_cast<double>(std::count_if(
```

Comment:
It might be useful to take the opportunity to rename the data members of STestStats to something more descriptive?

Reply:
Lovely, thanks Tom